lib/phoenix/live_dashboard/system_info.ex

```diff
@@ -235,10 +244,103 @@ defmodule Phoenix.LiveDashboard.SystemInfo do
     end
   end
 
+  ## Socket callbacks
+
+  def sockets_callback(search, sort_by, sort_dir, limit) do
+    sorter = if sort_dir == :asc, do: &<=/2, else: &>=/2
+
+    sockets =
+      for port <- Port.list(), info = socket_info(port), show_socket?(info, search), do: info
+
+    count = length(sockets)
+
+    sockets =
+      sockets
+      |> Enum.sort_by(&Keyword.fetch!(&1, sort_by), sorter)
+      |> Enum.take(limit)
+
+    {sockets, count}
+  end
+
+  def socket_info_callback(port, keys) do
+    case socket_info(port) do
+      :error -> :error
+      info -> Keyword.take(info, keys)
+    end
+  end
+
+  defp socket_info(port) do
+    with info when not is_nil(info) <- Port.info(port),
+         {:ok, stat} <- :inet.getstat(port, [:send_oct, :recv_oct]),
+         {:ok, state} <- :prim_inet.getstatus(port),
+         {:ok, {_, type}} <- :prim_inet.gettype(port),
+         module <- inet_module_lookup(port) do
+      info
+      |> Keyword.put(:port, port)
+      |> Keyword.merge(stat)
+      |> Keyword.merge(
+        module: module,
+        local_address: format_address(:inet.sockname(port)),
+        foreign_address: format_address(:inet.peername(port)),
+        state: format_socket_state(state),
+        type: type
+      )
+    else
+      _ -> :error
+    end
+  end
+
+  defp show_socket?(:error, _search), do: false
```

Comment:
If you return nil instead, the comprehension will remove it for you, so you don't need a clause in show_socket?.

```suggestion
      _ -> nil
    end
  end

```